x-pack/test/case_api_no_sub_cases_integration/basic/tests/cases/comments/find_comments.ts

```diff
@@ -0,0 +1,126 @@
+  const supertest = getService('supertest');
+  const es = getService('es');
+
+  describe('find_comments', () => {
+    afterEach(async () => {
+      await deleteCases(es);
+      await deleteComments(es);
+      await deleteCasesUserActions(es);
+    });
+
+    it('should find all case comment', async () => {
+      const { body: postedCase } = await supertest
+        .post(CASES_URL)
+        .set('kbn-xsrf', 'true')
+        .send(postCaseReq)
+        .expect(200);
+
+      // post 2 comments
+      await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postCommentUserReq)
+        .expect(200);
+
+      const { body: patchedCase } = await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postCommentUserReq)
+        .expect(200);
+
+      const { body: caseComments } = await supertest
+        .get(`${CASES_URL}/${postedCase.id}/comments/_find`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(200);
+
+      expect(caseComments.comments).to.eql(patchedCase.comments);
+    });
+
+    it('should filter case comments', async () => {
+      const { body: postedCase } = await supertest
+        .post(CASES_URL)
+        .set('kbn-xsrf', 'true')
+        .send(postCaseReq)
+        .expect(200);
+
+      // post 2 comments
+      await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postCommentUserReq)
+        .expect(200);
+
+      const { body: patchedCase } = await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send({ comment: 'unique', type: CommentType.user })
+        .expect(200);
+
+      const { body: caseComments } = await supertest
+        .get(`${CASES_URL}/${postedCase.id}/comments/_find?search=unique`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(200);
+
+      expect(caseComments.comments).to.eql([patchedCase.comments[1]]);
+    });
+
+    it('unhappy path - 400s when query is bad', async () => {
+      const { body: postedCase } = await supertest
+        .post(CASES_URL)
+        .set('kbn-xsrf', 'true')
+        .send(postCaseReq)
+        .expect(200);
+
+      await supertest
+        .post(`${CASES_URL}/${postedCase.id}/comments`)
+        .set('kbn-xsrf', 'true')
+        .send(postCommentUserReq)
+        .expect(200);
+
+      await supertest
+        .get(`${CASES_URL}/${postedCase.id}/comments/_find?perPage=true`)
+        .set('kbn-xsrf', 'true')
+        .send()
+        .expect(400);
+    });
+
+    describe('sub case comments', () => {
+      afterEach(async () => {
+        await deleteAllCaseItems(es);
+      });
+
+      it('fails to find comments for a sub case', async () => {
+        const { body } = await supertest
+          .get(`${CASES_URL}/invalid-id/comments/_find?subCaseId=somevalue`)
```

Comment:
Maybe I am missing something, but I cannot see how this test test the scenario described. We don't have comments, `subCaseId=somevalue` does not exist so the total will always be `0`.

Reply:
Good point, I have the routes through now 👍